packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,820 @@
+    transactionOverrides = {},
+  }: CreateSplitV2Config): Promise<TransactionFormat> {
+    recipients.map((recipient) => validateAddress(recipient))
+    validateAddress(owner)
+    creator = creator ?? zeroAddress
+    validateAddress(creator)
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const totalAllocation = allocations.reduce((a, b) => a + b)
+    const functionName = salt ? 'createSplitDeterministic' : 'createSplit'
+    const functionArgs = [
+      { recipients, allocations, totalAllocation, distributionIncentive },
+      owner,
+      creator,
+    ]
+    if (salt) functionArgs.push(salt)
+
+    return this._executeContractFunction({
+      contractAddress: getSplitV2FactoryAddress(this._chainId, splitType),
+      contractAbi: splitV2FactoryABI,
+      functionName,
+      functionArgs,
+      transactionOverrides,
+    })
+  }
+
+  protected async _transferOwnership({
+    split,
+    newOwner,
+    transactionOverrides = {},
+  }: TransferOwnershipConfig): Promise<TransactionFormat> {
+    validateAddress(split)
+    validateAddress(newOwner)
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    return this._executeContractFunction({
+      contractAddress: split,
+      contractAbi: splitV2ABI,
+      functionName: 'transferOwnership',
+      functionArgs: [newOwner],
+      transactionOverrides,
+    })
+  }
+
+  protected async _setPaused({
+    split,
+    paused,
+    transactionOverrides = {},
+  }: SetPausedConfig): Promise<TransactionFormat> {
+    validateAddress(split)
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    return this._executeContractFunction({
+      contractAddress: split,
+      contractAbi: splitV2ABI,
+      functionName: 'setPaused',
+      functionArgs: [paused],
+      transactionOverrides,
+    })
+  }
+
+  protected async _execCalls({
+    splitAddress,
+    calls,
+    transactionOverrides = {},
+  }: SplitV2ExecCallsConfig): Promise<TransactionFormat> {
+    validateAddress(splitAddress)
+    calls.map((call) => validateAddress(call.to))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    return this._executeContractFunction({
+      contractAddress: splitAddress,
+      contractAbi: splitV2ABI,
+      functionName: 'execCalls',
+      functionArgs: [calls],
+      transactionOverrides,
+    })
+  }
+
+  protected async _updateSplit({
+    split,
+    recipients,
+    allocations,
+    distributionIncentive,
+    transactionOverrides = {},
+  }: UpdateSplitV2Config): Promise<TransactionFormat> {
+    validateAddress(split)
+    recipients.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const totalAllocation = allocations.reduce((a, b) => a + b)
```

Comment:
nit: doesn't reduce need a starting value? e.g. `BigInt(0)`?

Reply:
if we don't specify an initial value it takes the first element of the array as the initial value